presto-main/src/main/java/io/prestosql/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -341,24 +341,29 @@ protected Node visitShowSchemas(ShowSchemas node, Void context)
         @Override
         protected Node visitShowCatalogs(ShowCatalogs node, Void context)
         {
-            List<Expression> rows = listCatalogs(session, metadata, accessControl).keySet().stream()
-                    .map(name -> row(new StringLiteral(name)))
-                    .collect(toList());
+            ImmutableList.Builder<Expression> rows = ImmutableList.builder();
 
-            Optional<Expression> predicate = Optional.empty();
+            listCatalogs(session, metadata, accessControl).keySet().stream()
+                    .map(name -> row(new StringLiteral(name), TRUE_LITERAL))
+                    .forEach(rows::add);
+
+            // add bogus row so we can support empty catalogs
+            rows.add(row(new StringLiteral(""), FALSE_LITERAL));
```

Comment:
We could simply filter out the empty row (since catalog names cannot be empty).

Reply:
If nothing is installed, there is a system catalog. However, AccessControl may legitimately filter it out. So it might result in a empty values

Reply:
I mean that the bogus row shouldn’t need a boolean column.

Reply:
Oh okay !!